Source/WebCore/accessibility/AXTextRun.cpp

```diff
@@ -173,19 +182,14 @@ FloatRect AXTextRuns::localRect(unsigned start, unsigned end, float lineHeight)
                     smallerOffset = offsetOfFirstCharacterInRun;
                 // Measure the characters in this run (accomplished by smallerOffset - offsetOfFirstCharacterInRun)
                 // prior to the offset.
-                unsigned widthPriorToStart = (smallerOffset - offsetOfFirstCharacterInRun) * estimatedCharacterWidth;
-                unsigned widthAfterEnd = runIndexOfSmallerOffset == runIndexOfLargerOffset
-                    // aa|aaa|aa
-                    // length 7, smallerOffset = 2, largerOffset = 5 — measure the last two "a" characters.
-                    ? (runLengthSumTo(i) - largerOffset) * estimatedCharacterWidth
-                    // The offsets pointed into different runs, so the width of this run extends to the end.
-                    : 0;
-                unsigned fullRunWidth = run.text.length() * estimatedCharacterWidth;
-
-                TEXT_RUN_ASSERT_AND_LOG(fullRunWidth >= (widthPriorToStart + widthAfterEnd), "localRect (2)");
-                if (fullRunWidth < (widthPriorToStart + widthAfterEnd))
-                    fullRunWidth = widthPriorToStart + widthAfterEnd;
-                measuredWidth = fullRunWidth - widthPriorToStart - widthAfterEnd;
+                unsigned widthPriorToStart = 0;
+                if (smallerOffset - offsetOfFirstCharacterInRun > 0)
+                    widthPriorToStart = computeAdvance(offsetOfFirstCharacterInRun, smallerOffset);
+
+                unsigned endOffsetInLine = runIndexOfSmallerOffset == runIndexOfLargerOffset ? largerOffset : !i ? run.text.length() : runLengthSumTo(i - 1) + run.text.length();
```

Comment:
Would it be too clunky to split this up into multiple lines rather than using a double-ternary? Might be easier to parse overall that way 🤔